BDArmory/Control/MissileFire.cs

```diff
@@ -4158,24 +4167,33 @@ public void UpdateList()
                     if (weapon.Current == null) continue;
                     string weaponName = weapon.Current.GetShortName();
                     bool alreadyAdded = false;
+                    
+                    bool currIsMissile = false;
```

Comment:
This section for missiles could be moved to after the weapon channel check to avoid going through it all if it's going to be skipped anyway. The only reason I can see for the gun section to be before the weapon channel check is for pure PD guns.